collect_app/src/main/java/org/odk/collect/android/spatial/MapHelper.java

```diff
@@ -320,4 +310,9 @@ private boolean isFileFormatSupported(File file) {
         }
         return result;
     }
+
+    // returns the index of selected map layer
```

Comment:
Thanks for addressing all the other comments!

I'd still like to suggest that two improvements to this description, though.

1. If you use `/** ... */` instead of `//`, then the comment will be interpreted as JavaDoc for this method.  That means that the description will appear in automatically generated documentation and will pop up automatically in Android Studio when someone hovers over the method or autocompletes it.

2. Because this is a public method, it's better to write this in a way that can be understood by someone who is a user of this class, not an implementor of it.  Someone who doesn't know the implementation won't be able to tell what "index of selected map layer" means—for example, does "index" mean an index into an array?  What is the array?  What does it mean for a map layer to be selected?  Does this imply that there are multiple layers displayed on top of each other?  We should try to write these is such a way as to avoid the possibility of such confusion.  Also, the description is incomplete—this method can return null.  What does it mean when it returns null?

Here's an example of how one might write a description for this method:

    /**
     * If the currently shown basemap is an offline layer, returns the index of the offline
     * layer's filename in the list of all the available filenames ending in ".mbtiles".
     * If the basemap is not an offline layer, returns null.
     */

Reply:
Thanks, I added a comment according to your example but with a small modification.
The modification is that I this method can return only integers, not nulls. Why because in the array of layers we always keep 'None' option at first place, so 0 means None and offline layers use indexes 1,2...etc. Maybe it's a little bit confusing but it would need more refactoring to change this approach so let's leave it as it is.

Reply:
Oh I'm really glad you updated the comment to make it correctly reflect
reality, then!  Clearly I misunderstood how it was supposed to work, so
explaining it in the comment helps a lot.  Thank you!